hugo/content/2-models/allowed-types.md

```diff
@@ -0,0 +1,173 @@
+                             MANGROVE_NVP(name),
+                             MANGROVE_NVP(hourly_wage),
+                             MANGROVE_NVP(contact_info))
+}
+```
+
+You can even include subdocuments in an STL container like this:
+
+```cpp
+class Employee : public mmangrove::model<Employee> {
+    std::string name;
+    double hourly_wage;
+    std::vector<ContactCard> contact_info;
+
+    MANGROVE_MAKE_KEYS_MODEL(Employee,
+                             MANGROVE_NVP(name),
+                             MANGROVE_NVP(hourly_wage),
+                             MANGROVE_NVP(contact_info))
+}
+```
+
+And the resulting document would look something like this:
+
+```json
+{
+    "_id" : ObjectId("578fcf796119b016e0a3b079"),
+    "name" : "Jenny",
+    "hourly_wage" : 19.81,
+    "contact_info" : [
+        {
+            "type" : "home",
+            "phone" : "212-867-5309",
+            "address" : "247 E 26 St, New York, NY"
+        },
+        {
+            "type" : "work",
+            "phone" : "555-123-4567",
+            "address" : "570 Second Ave, New York, NY"
+        }
+    ]
+}
+```
+
+The only restriction regarding serializable subdocuments is that they must be *default constructible*.
+
+## Optionals
+
+Although {{% a_blank "`std::optional<T>`" "http://en.cppreference.com/w/cpp/utility/optional/optional" %}} won't be available until C++17, Mangrove offers it as a serializable type in the form of `bson_mapper::stdx::optional<T>`. `optional` is a powerful new C++ type that specifies a value that may or may not exist.
+
+Mangrove offers support for optional types where `T` is a type discussed on this page. This gives Mangrove support for dynamic, flexible schemas, which will be discussed in a [later section](/2-models/dynamic-schemas).
+
+## ObjectIDs
+
+One of the fundamental types in MongoDB is the {{% a_blank "ObjectID" "https://docs.mongodb.com/manual/reference/bson-types/#objectid" %}}. ObjectIDs are the default type used for the `_id` field in documents, so you may want to use them in your project.
+
+The MongoDB C++ Driver offers the {{% a_blank "`bsoncxx::oid`" "http://mongodb.github.io/mongo-cxx-driver/classbsoncxx_1_1oid.html" %}}, which is a simple class that can be used to create and compare ObjectIDs. Mangrove offers serialization support for `bsoncxx::oid` so you can include them in your models as well. 
+
+## BSON Types
+
+The MongoDB C++ Driver offers a library called `libbsoncxx` that contains a set of utilities for using and manipulating BSON in C++. One of these utilities is a set of C++ types that cover every supported type in the {{% a_blank "BSON Specification" "http://bsonspec.org/spec.html" %}}.
+
+{{% notice warning %}}
+While the following types are all technically supported, we do not recommend using them if you can accomplish the same result with one of types discussed in the other sections. Many of the following types are dangerous and only used internally by MongoDB.
```

Comment:
> Many of the following types are dangerous and only used internally by MongoDB.

Definitely gets the point across :) Can you also add something like "To learn more about these types read <bsonspec_link> and <bsconcxx_whatever_file_these_are_in_link>"